### PR TITLE
Fix missing symbol in sql-py on osx-arm64

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,6 +29,7 @@ requirements:
     - setuptools_scm_git_archive
     - libtiledb-sql >=0.10
     - libxcrypt                             # [linux]
+    - mysql-devel
```
Comment (Shelnutt2) [inline recipe/meta.yaml:32]: I think this is still problematic. Everything should come from this package only.

Comment (DimitrisStaratzis) [reply recipe/meta.yaml:32]: Yeah, it is like this in the mysqlclient-feedstock, hence my try:

https://github.com/conda-forge/mysqlclient-feedstock/blob/main/recipe/meta.yaml

**Comment by jdblischak** [comment]
What is the motivation for this PR? The [CI](https://github.com/conda-forge/libtiledb-sql-py-feedstock/runs/21780763674) on the latest commit, b9106fa, passed

**Comment by DimitrisStaratzis** [comment]
> What is the motivation for this PR? The [CI](https://github.com/conda-forge/libtiledb-sql-py-feedstock/runs/21780763674) on the latest commit, [b9106fa](https://github.com/conda-forge/libtiledb-sql-py-feedstock/commit/b9106faef1de032aa5700bd72d684481b5ae83d6), passed

https://app.shortcut.com/tiledb-inc/story/41361/missing-symbol-in-sql-py-on-osx-arm64

**Comment by DimitrisStaratzis** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (DimitrisStaratzis) [reply recipe/meta.yaml]: We don't need it you are right.

**Comment by DimitrisStaratzis** [comment]
suppressed by #29

**Pull request closed without merging**